File: backend/fastapi/app/domain/sport.py

from datetime import datetime
from enum import Enum
from typing import Optional
from pydantic import BaseModel
from app.domain.enums.general_status import GeneralStatus

class Sport(BaseModel):
    id: int
    slug: str
    name: str
    description: Optional[str] = None
    img_url: Optional[str] = None
    status: GeneralStatus
    is_active: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: backend/policourt-backend/src/main/java/com/policourt/api/sport/domain/model/Sport.java
package com.policourt.api.sport.domain.model;

import java.time.OffsetDateTime;

import com.policourt.api.shared.enums.GeneralStatus;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Data
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Sport {
    private Long id;
    private String slug;
    private String name;
    private String description;
    private String imgUrl;
    private GeneralStatus status;
    private boolean isActive;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}